src/app/navigation/middleware.ts

```diff
@@ -28,6 +30,11 @@ export default (routes: AnyRoute[], history: History): Middleware => ({dispatch}
       return;
     }
 
+    action.payload.search = queryString.stringify({
```

Comment:
you're stringifying this just so `matchSearch` can immediately parse it again. the second argument to `matchSearch` is only used from here, so you could change the way it works to be less silly